### PR TITLE
ci: introduce helm snapshots and ci verification

### PR DESCRIPTION
**Pull request by lukasmetzner**
Introduces snapshots generated with helm template from default values, example production values and full values. A script to generate this snapshots is run in the CI with a check if they are up to date.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -14,13 +14,14 @@ repos:
 
       - id: check-json
       - id: check-yaml
-        exclude: ^(deploy/.*|chart/templates/.*)$
+        exclude: ^(deploy/.*|chart/templates/.*|chart/.snapshots/.*)$
```
Comment (apricote) [inline .pre-commit-config.yaml:17]: I do think it makes sense to validate the generated snapshots for valid yaml. 

What was the problem that made you exclude the snapshots?

File: chart/example-prod.values.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+replicaCount: 3
```
Comment (apricote) [inline chart/example-prod.values.yaml:1]: I dont think the prod files add any values here. The full template already does nearly everything we can do.

File: chart/.snapshots/full.values.yaml [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+  # https://github.com/hetznercloud/hcloud-cloud-controller-manager/issues/395
+  route-reconciliation-period: 30s
+
+  # We do not use the webhooks feature and there is no need to bind a port that is unused.
+  # https://github.com/kubernetes/kubernetes/issues/120043
+  # https://github.com/hetznercloud/hcloud-cloud-controller-manager/issues/492
+  webhook-secure-port: "0"
+
+# Change deployment kind from "Deployment" to "DaemonSet"
+kind: Deployment
```
Comment (apricote) [inline chart/.snapshots/full.values.yaml:20]: What do you think about changing this to `DaemonSet` so we have one snapshot for each?

Comment (lukasmetzner) [reply chart/.snapshots/full.values.yaml]: Sounds good. Changed the type to DaemonSet for `full.values.yaml`.

**Review by apricote — APPROVED** [review]

**Review by apricote — CHANGES_REQUESTED** [review]
CI fails because of the file rename.

**Review by apricote — APPROVED** [review]

**Pull request merged**